pkg/operation/botanist/addons.go

```diff
@@ -243,7 +244,7 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 			"blackbox-exporter": blackboxExporter,
 		}, b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting),
 		"network-policies":        networkPolicyConfig,
-		"podsecuritypolicies":     common.GenerateAddonConfig(podSecurityPolicies, true),
+		"podsecuritypolicies":     common.GenerateAddonConfig(podSecurityPolicies, !b.Shoot.PSPDisabled),
```

Comment:
I think the nginx PSP rbac should be also adapted.
https://github.com/gardener/gardener/blob/02b3d19475722016d0d4dae010b655849b6e1e79/charts/shoot-addons/charts/nginx-ingress/templates/psp/nginx-ingress-rolebinding-privileged.yaml#L9

The `gardener.cloud:psp:privileged` ClusterRole won't be present when the PodSecurityPolicy admission is disabled, right?

There are also 2 tm machinery tests that rely on the `gardener.cloud:psp:privileged` ClusterRole. Maybe they can also adapted.